### ISSUE TITLE
hwloc-2.0 support

### ISSUE THREAD

**Issue by keszybz** [StanfordLegion/legion#575]
Hi,
mpich 3.3 now requires hwloc>=2.0. It would be great if legion could be updated too.
This was already signalled in #479 in February :(

**Comment by streichler** [comment]
Yes, this is somewhere on our task list.  It's not super-high though, as you can almost
certainly disable hwloc in the Legion/Realm build with no loss of functionality.  (Or
conversely, if you have a system that the linux-sysfs-based discovery code doesn't work
as well as hwloc for, let me know and we'll consider bumping this up the priority list.)

**Comment by keszybz** [comment]
Thanks for the quick reply. I made a small patch to fix one of the two compilation issues.
I also did the suggested thing and disabled the use of hwloc that does not compile...

> Or conversely, if you have a system that the linux-sysfs-based discovery code doesn't work
as well as hwloc for, let me know and we'll consider bumping this up the priority list.

I wish I knew ;) I'm just the packager here, and I'm only touching legion because I want to update mpich.

**Comment by streichler** [comment]
As mentioned in the pull request, I'd like this change to add hwloc-2.x support without losing hwloc-1.x support, so it'll all need to be ifdef-protected.

**Comment by scardracs** [comment]
Still waiting for update to hwloc-2.6.0

**Comment by scardracs** [comment]
If you have any patch please provide it as I could add it to gentoo ebuild

**Comment by streichler** [comment]
@scardracs As was suggested a while back, I think the simplest thing to do here is to remove the Legion package's dependence on hwloc entirely.  We routinely build without hwloc in our development, so I'm not sure what the issue that @keszybz encountered before was.  If you're not sure how to disable use of hwloc in Legion, please point me at your build script and I'll see if I can suggest something.

**Comment by scardracs** [comment]
> @scardracs As was suggested a while back, I think the simplest thing to do here is to remove the Legion package's dependence on hwloc entirely.  We routinely build without hwloc in our development, so I'm not sure what the issue that @keszybz encountered before was.  If you're not sure how to disable use of hwloc in Legion, please point me at your build script and I'll see if I can suggest something.

Yeah I know how to remove it, the only point is that Gentoo ebuild use +hwloc, which means is forced by default. If you prefer to remove it I can proceed on dropping it on Gentoo side (as I'm working on cleanup/maintain it)

**Comment by streichler** [comment]
I'm not that familiar with Gentoo's build system, but having Legion's use of hwloc be disabled by default sounds better to me.

**Comment by scardracs** [comment]
> I'm not that familiar with Gentoo's build system, but having Legion's use of hwloc be disabled by default sounds better to me.

Short answer:

Gentoo uses ebuilds, which use a list of flags (IUSE). On legion there is a IUSE="... +hwloc ..." and that means it is forced by default if not disabled system-wide. I'm gonna drop it this evening.
